scripts/build_test.py

```diff
@@ -36,7 +107,197 @@ def test_insert_hash(self):
                           'path/to/file.min.fedcba.js')
         # pylint: enable=protected-access
 
+    def test_ensure_directory_exists(self):
+        """Test ensure_directory_exists to make sure non-existent parent
+        directory of random.js must be created after calling function.
+        """
+        # Setup a sandbox folder to assert for non-existence.
+        SANDBOX_DIR = os.path.join(build.PARENT_DIR, 'sandbox')
+        random_filepath = os.path.join(SANDBOX_DIR, 'random.js')
+        # Asserting ../oppia/random does not exist.
+        self.assertFalse(os.path.isdir(SANDBOX_DIR))
+        build.ensure_directory_exists(random_filepath)
+        # Asserting ../oppia/random exists.
+        self.assertTrue(os.path.isdir(SANDBOX_DIR))
+        # Clean up empty sandbox directory that has just been created.
+        shutil.rmtree(SANDBOX_DIR)
+
+    def test_ensure_files_exist(self):
+        """Test _ensure_files_exist raises exception with a non-existent
+        filepath.
+        """
+        random_filepaths = [
+            os.path.join(build.THIRD_PARTY_GENERATED_DEV_DIR, 'random1.js')]
+        # pylint: disable=protected-access
+        with self.assertRaises(ValueError) as fileNotExist:
+            build._ensure_files_exist(random_filepaths)
+        # pylint: enable=protected-access
+        # Exception will be raised at first file determined to be non-existent.
+        self.assertTrue(
+            ('File %s does not exist.') % random_filepaths[0] in
+            fileNotExist.exception)
+
+    def test_get_file_count(self):
+        """Test get_file_count to return the correct number of files, minus
+        ignored files.
+        """
+        all_inclusive_file_count = 0
+        for _, _, files in os.walk(build.EXTENSIONS_DIR.get('dev_dir')):
+            all_inclusive_file_count += len(files)
+        ignored_file_count = 0
+        for _, _, files in os.walk(build.EXTENSIONS_DIR.get('dev_dir')):
+            for filename in files:
+                if any(filename.endswith(p)
+                       for p in build.FILE_EXTENSIONS_TO_IGNORE):
+                    ignored_file_count += 1
+        self.assertEqual(
+            all_inclusive_file_count - ignored_file_count,
+            build.get_file_count(build.EXTENSIONS_DIR.get('dev_dir')))
+
+    def test_compare_file_count(self):
+        """Test _compare_file_count to raise exception when there is a
+        mismatched file count between 2 dirs.
+        """
+        with self.assertRaises(ValueError) as incorrectFileCount:
+            # pylint: disable=protected-access
```

Comment:
Mmm -- this is not usual, but given the prevalence of this pragma in the file, and the fact that I think the build functions _should_ be private, let's just have this pragma on the top of the file and the enable on the bottom. And add a comment to explain why.

I think that would make the tests more readable.

Reply:
Overrode pylint rule at the beginning and re-enabled at the end of file.
Also added explanation comment!